### ISSUE TITLE
Calling remote webpage JavaScript through webview.openWebViewURL()

### ISSUE THREAD

**Issue by aammfe** [myflashlab/webView-ANE#28]
Hi!
    when i call javaScript with
 webview.openWebViewURL(0 ,0 , stage.stageWidth , stage.stageHeight - 100,"javaScript:alert('hi')");

![image](https://github-cloud.s3.amazonaws.com/assets/11896817/10566059/47b877c2-7591-11e5-9685-6801b3134fd5.png)

previous page dispose or hide , and then show alert 

![image](https://github-cloud.s3.amazonaws.com/assets/11896817/10566141/7ecfad46-7593-11e5-8e54-9d30c03df32e.png)

but when i use stageWebView 
flashWebView.loadURL("javaScript:alert('hi')");

![image](https://github-cloud.s3.amazonaws.com/assets/11896817/10566085/efe04218-7591-11e5-8ced-daaac8ad2c71.png)

previous page donot hide or destroy.

![image](https://github-cloud.s3.amazonaws.com/assets/11896817/10566076/b2235a8c-7591-11e5-869c-46489c1178bd.png)

In other words i want to call javascript of remote website to acess its html. 
  
i m student and date of my final year project is near to come , plz provide solution as soon as possible.
       Thanx

**Comment by myflashlab** [comment]
with our RichWebview ANE, you can only open a webpage (local or online) with commands like ```openWebViewURL``` or ```openWebViewLocal```. In future updates we will add html content scripts to be open, but for now, you can only open an actual html or a web content page.

Considering that RichWebview has a complete solution for flash/JS communication, you actually don't need to do things the old way at all. if you need your flash side to open an alert window on the JS side, you can simply call the JS function from the flash side.

use the ```openWebViewURL``` and ```openWebViewLocal``` methods only when you are trying to open a page.

**Comment by aammfe** [comment]
JS function is just for local webpage or page which have 
<script type="text/javascript" src="Bridge.js"></script>

what if we have to interact with remote webpages which do not have
 <script type="text/javascript" src="Bridge.js"></script>....?

kindly explain it with example ...

**Comment by myflashlab** [comment]
no matter if your html content is local or remote, you can always communicate between flash/JS using the Bridge.js

you can't make a successful connection on webpages which do not have the Bridge.js and that makes total sense because if you are planing to show html content on your app and you wish to communicate with it via your Air app, then it means you are building the html content yourself. so you should be able to implement the bridge.js in it. but if you are visiting a third-party webpage, then you can't expect their webpage to interact with your app!

**Comment by aammfe** [comment]
"but if you are visiting a third-party webpage, then you can't expect their webpage to interact with your app!"

is there any way we can interact with 3rd party websites ?
because in real (chrome and firefox ) you can insert your script at console , (javaScript injection ) is there any way to do this type of thing in your extensions. with out this bridge ..?

what if every Web-side that opens in your extension by Default have "bridge"  script ..?
means every site can interact ....

**Comment by myflashlab** [comment]
I totally understand your point but as we mentioned earlier, you cannot inject JS at the moment. we are planned to add this feature in future versions. RichWebView has been born for only a few months and we are constantly adding features to it but we are making sure to add the "most" used features first. JS injection will be added soon too :)

**Comment by aammfe** [comment]
plz release next version asap because my final year project has been stuck because of this feature , and date of submition  is 28 and after this feature has been added , there should be time to develop my app and 95 % of my project just depend upon your product .

**Comment by myflashlab** [comment]
We hope you can do the best in your final year project. about the release dates, we have to do things according to our own timetable but you can always contact us for a custom job order if you need a feature sooner.

**Comment by aammfe** [comment]
when next version is going to release (including "bridge or javaScript injection feature") ...?

**Comment by myflashlab** [comment]
injection feature is already added in V5 or greater https://github.com/myflashlab/webView-ANE/issues/48
